x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -567,19 +570,27 @@ public static boolean isSpatialPoint(DataType t) {
     }
 
     public static boolean isSpatialShape(DataType t) {
-        return t == GEO_SHAPE || t == CARTESIAN_SHAPE;
+        return t == GEO_SHAPE || t == CARTESIAN_SHAPE || t == GEOHASH || t == GEOTILE || t == GEOHEX;
     }
 
     public static boolean isSpatialGeo(DataType t) {
-        return t == GEO_POINT || t == GEO_SHAPE;
+        return t == GEO_POINT || t == GEO_SHAPE || t == GEOHASH || t == GEOTILE || t == GEOHEX;
     }
 
     public static boolean isSpatial(DataType t) {
         return t == GEO_POINT || t == CARTESIAN_POINT || t == GEO_SHAPE || t == CARTESIAN_SHAPE;
     }
 
+    public static boolean isSpatialOrGrid(DataType t) {
+        return isSpatial(t) || isGeoGrid(t);
+    }
+
+    public static boolean isGeoGrid(DataType t) {
+        return t == GEOHASH || t == GEOTILE || t == GEOHEX;
+    }
+
     public static boolean isSortable(DataType t) {
-        return false == (t == SOURCE || isCounter(t) || isSpatial(t) || t == AGGREGATE_METRIC_DOUBLE);
+        return false == (t == SOURCE || isCounter(t) || isSpatialOrGrid(t) || t == AGGREGATE_METRIC_DOUBLE);
     }
```

Comment:
Are the grid types supposed to be sortable?

I think they shouldn't be, right? Maybe a small verifier test would help ensure that the validation of `SORT` and `SORT` triggers and flags the grid data types as invalid.

Reply:
Correct, they are not sortable. If a user wishes to sort, they can do `TO_STRING` first and sort on that, which makes it clear what they are sorting on. I can look at making a validation test for this.